aten/src/ATen/core/TensorImpl.h

```diff
@@ -1410,10 +1469,15 @@ struct CAFFE2_API TensorImpl : public c10::intrusive_ptr_target {
     is_contiguous_ = compute_contiguous();
   }
 
-public:
-  at::Storage storage_; // TODO: Fix visibility on me
-
 protected:
+  at::Storage storage_;
+  // This pointer points to an AutogradMeta struct that stores autograd-specific fields
+  // (such as grad_ / grad_fn_ / grad_accumulator_). It is declared as a void* pointer
+  // because we don't want to expose autograd types to libATen.
+  // This pointer always has unique ownership (meaning only one TensorImpl can own it
+  // at a time).
+  void* autograd_meta_ = nullptr;
```

Comment:
should this be a unique_ptr then?  @ezyang any opinion?

Reply:
it seems that we could use `unique_ptr<void>` if we also pass a custom deleter to the `unique_ptr` (according to https://stackoverflow.com/questions/39288891/why-is-shared-ptrvoid-legal-while-unique-ptrvoid-is-ill-formed). This will make the unique ownership more easily maintained, but it also adds one more word to the size of a TensorImpl object, which we might not want.